src/AutoConstructor.Generator/Models/FieldInfo.cs

```diff
@@ -1,15 +1,13 @@
-using Microsoft.CodeAnalysis;
+namespace AutoConstructor.Generator.Models;
 
-namespace AutoConstructor.Generator;
-
-public class FieldInfo
+internal sealed record FieldInfo
```

Comment:
How come you're not just using the normal record syntax here with a primary constructor? Would be much shorter.